config/env/env_types.go

```diff
@@ -0,0 +1,161 @@
+package env
+
+import (
+	"errors"
+	"os"
+	"regexp"
+	"strconv"
+)
+
+var TypeError = errors.New("type could not be converted")
+var EnvironmentError = errors.New("environment variable not found")
+
+var regex *regexp.Regexp
+
+func init() { regex = regexp.MustCompile(`\${[A-Z]+[A-Z1-9_]*}`) }
+
+func replaceEnvVar(in string) (string, error) {
+
+	// TODO(ear7h): FindAllString
+	varName := regex.FindString(in)
+	// no env var
+	if len(varName) == 0 {
+		return in, nil
+	}
+
+	// trim the leading '${' and trailing '}'
+	varName = varName[2:len(varName)-1]
+
+	// get env var
+	envVar := os.Getenv(varName)
+	if envVar == "" {
```

Comment:
use [os.LookupEnv](https://godoc.org/os#LookupEnv) here instead of `os.GetEnv`. You should only return this error if the environment var was not defined. Defining it to an empty value may be correct.